### PR TITLE
Fix: ensure selectable prices is not 0 before applying filter

### PR DESCRIPTION
**Pull request by SWH-Relewise**

### CODE REVIEW

File: src/components/SearchOverlay.vue [hunk 1]
```diff
@@ -84,7 +84,12 @@ async function search() {
     let applySalesPriceFacet = false;
     if (result.value?.facets?.items?.length === 3) {
         const salesPriceFacet = result.value?.facets.items[2] as PriceRangeFacetResult;
-        applySalesPriceFacet = salesPriceFacet && filters.value.price.length === 2 && Number(filters.value.price[0]) !== salesPriceFacet.available!.value?.lowerBoundInclusive || Number(filters.value.price[1]) !== salesPriceFacet.available!.value?.upperBoundInclusive;
+        applySalesPriceFacet = salesPriceFacet
+            && filters.value.price.length === 2
+            && (Number(filters.value.price[0]) !== salesPriceFacet.available!.value?.lowerBoundInclusive
+                && salesPriceFacet.available!.value?.lowerBoundInclusive !== 0)
+            || (Number(filters.value.price[1]) !== salesPriceFacet.available!.value?.upperBoundInclusive
+                && salesPriceFacet.available!.value?.upperBoundInclusive !== 0);
```
Comment (mzanoni) [inline src/components/SearchOverlay.vue:92]: Sweet, but can we do something to make the code a bit more readable?

Comment (SWH-Relewise) [reply src/components/SearchOverlay.vue]: Check! I have split each check out into its own variable.

File: src/components/SearchOverlay.vue [hunk 2]
```diff
@@ -84,7 +84,16 @@ async function search() {
...
+        
+        const bothPriceFiltersSet = filters.value.price.length === 2;
+
+        const lowerBoundNotEqualOrZero = (Number(filters.value.price[0]) !== salesPriceFacet.available!.value?.lowerBoundInclusive
+                && salesPriceFacet.available!.value?.lowerBoundInclusive !== 0);
+
+        const upperBoundNotEqualOrZero = (Number(filters.value.price[1]) !== salesPriceFacet.available!.value?.upperBoundInclusive
+                && salesPriceFacet.available!.value?.upperBoundInclusive !== 0);
+
+        applySalesPriceFacet = salesPriceFacet && bothPriceFiltersSet && lowerBoundNotEqualOrZero || upperBoundNotEqualOrZero;
```
Comment (mzanoni) [inline src/components/SearchOverlay.vue:96]:
```suggestion
        applySalesPriceFacet = salesPriceFacet && bothPriceFiltersSet && (lowerBoundNotEqualOrZero || upperBoundNotEqualOrZero);
```

**Review by mzanoni — COMMENTED** [review]
I love it! I just have small OCD thingy about lines with multiple and/or :|

File: src/components/SearchOverlay.vue (hunk 2 above)
Comment (SWH-Relewise) [reply src/components/SearchOverlay.vue]: You got it!

**Review by mzanoni — APPROVED** [review]

**Pull request merged**